spec/policies/division_policy_spec.rb

```diff
@@ -10,27 +10,41 @@
   context 'being a member of a division' do
     let(:user) { create(:user, :member, division: division) }
 
-    permit_actions [:index, :show]
+    it 'can not index' do
+      expect(described_class.new(user, Division).index?).to be_falsey
+    end
+    permit_actions [:show]
     forbid_actions [:create, :edit, :update, :destroy]
   end
 
   context 'being an admin of a division' do
     let(:user) { create(:user, :admin, division: division) }
 
-    permit_all
+    it 'can index' do
+      expect(described_class.new(user, Division).index?).to be_truthy
+    end
+    permit_actions [:show, :edit, :update]
+    forbid_actions [:create, :destroy]
   end
 
   context 'being a member of a parent division' do
     let(:user) { create(:user, :member, division: parent_division) }
 
-    permit_actions [:index, :show]
+    permit_actions [:show]
```

Comment:
This is fine.

On Fri, May 6, 2016 at 11:48 AM, Joseph Elliott <notifications@github.com>
wrote:

> In spec/policies/division_policy_spec.rb
> <https://github.com/sassafrastech/madeline_system/pull/76#discussion_r62348458>
> :
>
> >    end
> >
> >    context 'being a member of a parent division' do
> >      let(:user) { create(:user, :member, division: parent_division) }
> >
> > -    permit_actions [:index, :show]
> > +    permit_actions [:show]
>
> I'm only testing the index permission for the two 'division' contexts.
> Since the parent/child relations doesn't impact. Would you want duplicate
> tests included for the other permutations anyways?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/76/files/b996bf238caf158080fcee1cfa09858482b64ce3#r62348458>
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org